Utility/WikiObjectPropertiesListGenerator.FCMacro

```diff
@@ -0,0 +1,401 @@
+}
+
+
+def clear_report_view():
+    """Clear content of Report View panel."""
+    main_window = Gui.getMainWindow()
+    main_window.findChild(QtGui.QTextEdit, "Report view").clear()
+
+
+def print_sr(text: str, show_report: bool = True):
+    """Print the given text if "show report" flag is enabled."""
+    FreeCAD.Console.PrintMessage(text) if show_report else None
+
+
+def generate_markup_list(grouped_properties: dict, print_status: bool = False):
+    """Print to the console the Wiki markup code."""
+    # NOTE: Group example:
+    # {{Properties_Title|Box features}}
+    # Property example:
+    # * {{PropertyData|BoxOpenFront|Bool}}: Default is {{value|False}}. Leave front of box open
+    # or:
+    # * {{PropertyData|BoxOpenFront|Bool|Dynamic}}: Default is {{value|False}}. Leave front of box open
+
+    # Sort groups and properties within each group
+    for group in sorted(grouped_properties):
+        FreeCAD.Console.PrintMessage(f"\n{{{{Properties_Title|{group}}}}}\n\n")
+        for prop, data in sorted(grouped_properties[group]):
+            if print_status:
+                status = "|" + data.status
+            elif "Hidden" in data.status:  # make exception, all other status will be ignored
+                status = "|" + "hidden"  # manually use lowercase
+            else:
+                status = ""
+            FreeCAD.Console.PrintMessage(
+                f"* {{{{Property{data.property_type}|{prop}|{data.data_type}{status}}}}}: "
+                f"Default is {{{{value|{data.default_value}}}}}. {data.tooltip}.\n"
+            )
+    FreeCAD.Console.PrintMessage("\n\n")
+
+
+def print_obj_properties(
+    obj, typ: str, ignore_list: dict, show_inherits: bool = False, print_status: bool = False
+):
+    """Print non-ignored properties of the selected object to the Report View."""
+    grouped_properties = {}
+    inherits = ""
+    for prop in obj.PropertiesList:
+        value = getattr(obj, prop, "N/A")
+        typeid = (obj.getTypeIdOfProperty(prop) or "").replace("App::Property", "")
+        prop_group = obj.getGroupOfProperty(prop) or ""
+        tooltip = obj.getDocumentationOfProperty(prop) or ""
+        # Get all possible status variants
+        stat = obj.getPropertyStatus(prop) or ""
+
+        if stat != (""):
+            for st in stat:
+                # get status without name from dictionary
+                if isinstance(st, int):
+                    stat.insert(stat.index(st), status_dictionary[st])
+                    stat.remove(st)
+            # remove symbols ['] from str by regexp and trim spaces
+            stat = re.sub(r"[\[\]\']", "", str(stat)).strip()
+            # NOTE: Uncomment next two lines to make the list look like the list of parameters from the editor
+            # if 'Hidden' in stat:
+            #   continue
+        else:
+            stat = ""
+
+        if ignore_list is None or (ignore_list is not None and prop not in ignore_list):
```

Comment:
According to the typing hint and how you call `print_obj_properties`, `ignore_list` is never supposed to be `None`. You can simplify the condition.